src/main/logic/Logic.cpp

```diff
@@ -108,13 +109,24 @@ void Logic::collisionDetection() {
 	//Scan for player collisions
 	EntityList::iterator iter;
 	for (PlayerList::iterator it = allPlayers.begin(); it != allPlayers.end(); it++) {
```

Comment:
Now that we're using C++ 11, we can use a ranged-based for loop here to make the iterator handling a bit easier.

So we'd have 'for (PlayerShPtr player : allPlayers) { ... } ' and you can then use 'player->foo()' instead of having to dereference the iterator manually.